Tactics/Sym/LCtxSearch.lean

```diff
@@ -95,9 +97,14 @@ variable {m}
   def-eq to the pattern, or if `whenFound` returned `skip` for all variables
   that were found. For convenience, we pass in the `expectedType` here as well.
   See `throwNotFound` for a convenient way to throw an error here.
-- If `changeType` is set to true, then we change the type of every successful
-  match (i.e., `whenFound` returns `continu` or `done`) to be exactly the
-  `expectedType`
+- If `changeType` (default `false`) is set to true, then we change the type of
```

Comment:
@alexkeizer Please change the `SymContext` to also set `matchUnderBinders` to true to pick up on non-effects.

Reply:
In hindsight, this isn't as straightforward: when constructing a `SymContext` we have that the initial state and current state are the same, so a non-effect hyp would look like `forall f, f ≠ _ -> r f <state> = r f <state>`. This is, of course, a tautology (and the default thing we initialize the non-effects proof to), so there's no need to search for it.

We *could* search for `forall f, f ≠ _ -> r f <state> = r f ?newInitialState`, where the rhs refers to an mvar, and update the initial state accordingly. This gets into tricky territory, because we then have to make sure that all of the fields mentioned in the sequence of non-equalities have a value assigned to them in the other proofs. 

In conclusion, let's not tackle this particular problem in this PR. We can just keep `sym_aggregate` around in proofs that need to aggregate non-effects, and think more about this after we've solved intermediate state aggregation (since that's when having the initial state populated correctly becomes more important). For now, we can always work-around this by having explicit hypotheses `r x12 s1 = r x12 s0` hypotheses for those fields that we care about being tracked, as a workaround.